test/Concurrency/async_main.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: %target-swift-frontend -dump-ast -enable-experimental-concurrency -parse-as-library %s | %FileCheck %s --check-prefix=CHECK-AST
+// REQUIRES: concurrency
+
+func asyncFunc() async { }
+
+@main struct MyProgram {
+  static func main() async {
+    await asyncFunc()
+  }
+}
+
```

Comment:
Perhaps create another test (if it can't be in this file) where main is `async throws`, to cover that scenario too? Also, I wonder how the implementation of `_runAsyncMain` takes care of unhandled exceptions, since `runAsyncAndBlock`'s type doesn't say that it handles throwing functions.

Reply:
Sounds good. It quietly swallows the error. It should probably be consistent with `runDetatched` and the non-async main and explode violently.

Reply:
Okay, we explode properly now.